File: IPWorks ZIP Samples/Open SevenZip/java/opensevenzip.java
/*
 * IPWorks ZIP 2024 Java Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks ZIP in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworkszip
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */

import java.io.*;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import ipworkszip.*;

public class opensevenzip extends ConsoleDemo {
	SevenZip sevenzip;
	String buffer;
	public opensevenzip(){
		sevenzip = new SevenZip();
		try{
			sevenzip.setArchiveFile(prompt("Please enter the name of the 7z file to extract",":","samplezip.7z"));
			sevenzip.setExtractToPath(prompt("Please enter the path for extraction",":","./"));
		  		  
			sevenzip.extractAll();
			System.out.println("Archive extracted.");
			System.out.println("\r\nPress enter to continue...");
			input();
			System.exit(0);			  
		}catch(IPWorksZipException ex){
			System.out.println("IPWorksZip exception thrown: " + ex.getCode() + " [" + ex.getMessage() + "].");
		}
		catch(Exception ex){
			System.out.println(ex.getMessage());
		}
	}
	
	public static void main(String[] args) {
	new opensevenzip();
	}
}


class ConsoleDemo {
  private static BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

  static String input() {
    try {
      return bf.readLine();
    } catch (IOException ioe) {
      return "";
    }
  }
  static char read() {
    return input().charAt(0);
  }

  static String prompt(String label) {
    return prompt(label, ":");
  }
  static String prompt(String label, String punctuation) {
    System.out.print(label + punctuation + " ");
    return input();
  }
  static String prompt(String label, String punctuation, String defaultVal) {
      System.out.print(label + " [" + defaultVal + "] " + punctuation + " ");
      String response = input();
      if (response.equals(""))
        return defaultVal;
      else
        return response;
  }

  static char ask(String label) {
    return ask(label, "?");
  }
  static char ask(String label, String punctuation) {
    return ask(label, punctuation, "(y/n)");
  }
  static char ask(String label, String punctuation, String answers) {
    System.out.print(label + punctuation + " " + answers + " ");
    return Character.toLowerCase(read());
  }

  static void displayError(Exception e) {
    System.out.print("Error");
    if (e instanceof IPWorksZipException) {
      System.out.print(" (" + ((IPWorksZipException) e).getCode() + ")");
    }
    System.out.println(": " + e.getMessage());
    e.printStackTrace();
  }
}

File: IPWorks ZIP Samples/Open SevenZip/javascript/opensevenzip.js
/*
 * IPWorks ZIP 2024 JavaScript Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks ZIP in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworkszip
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */
 
const readline = require("readline");
const ipworkszip = require("@nsoftware/ipworkszip");

if(!ipworkszip) {
  console.error("Cannot find ipworkszip.");
  process.exit(1);
}
let rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

main();

async function main() {

  const sevenzip = new ipworkszip.sevenzip();

  prompt('archiveFile', 'Please enter the name of the 7zip file to extract', ':', 'samplezip.7z');

  sevenzip.on('EndFile',(e) => {
    console.log('Extracted: ' + sevenzip.getFiles().get(e.index).getDecompressedName());
  });
  rl.on('line', async function(line) {
    switch (lastPrompt) {
      case 'archiveFile': {
        sevenzip.setArchiveFile(line === '' ? lastDefault : line);
        prompt('extractToPath', 'Please enter the path for extraction', ':', './');
        break;
      }
      case 'extractToPath': {
        sevenzip.setExtractToPath(line === '' ? lastDefault : line);
        console.log('Extracting...');;
        try {
          await sevenzip.extractAll();
          console.log('Archive extracted');
          process.exit();
        }
        catch (e) {
          console.log(e);
          process.exit();
        }
      }
    }
  });
}

function prompt(promptName, label, punctuation, defaultVal)
{
  lastPrompt = promptName;
  lastDefault = defaultVal;
  process.stdout.write(`${label} [${defaultVal}]${punctuation} `);
}
